src/analysis/retail/deathknight/unholy/modules/talents/PlagueBringer.tsx

```diff
@@ -0,0 +1,100 @@
+import { t } from '@lingui/macro';
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/TALENTS/deathknight';
+import { SpellLink } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { ApplyBuffEvent, RemoveBuffEvent, FightEndEvent } from 'parser/core/Events';
+import { When } from 'parser/core/ParseResults';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import { formatPercentage } from 'common/format';
+
+class PlagueBringer extends Analyzer {
+  lastBuffTime = 0;
+  totalBuffTime = 0;
+  buffIsUp = false;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS.PLAGUEBRINGER_TALENT);
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.PLAGUEBRINGER_BUFF),
+      this.onBuffEvent,
+    );
+
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.PLAGUEBRINGER_BUFF),
+      this.onRemoveBuffEvent,
+    );
+
+    this.addEventListener(Events.fightend, this.onFightEnd);
+  }
+
+  onBuffEvent(event: ApplyBuffEvent) {
+    this.lastBuffTime = event.timestamp;
+    this.buffIsUp = true;
+  }
+
+  onRemoveBuffEvent(event: RemoveBuffEvent) {
+    this.totalBuffTime += event.timestamp - this.lastBuffTime;
+    this.buffIsUp = false;
+  }
+
+  onFightEnd(event: FightEndEvent) {
+    if (this.buffIsUp === true) {
+      this.totalBuffTime += event.timestamp - this.lastBuffTime;
+    }
+  }
```

Comment:
instead of tracking the buff uptime yourself, you can use `this.selectedCombatant.getBuffUptime(SPELLS.PLAGUEBRINGER_BUFF.id)`

Reply:
Changed to make use of this instead!